application/client/src/app/ui/elements/recent/state.ts

```diff
@@ -78,6 +78,21 @@ export class State extends Holder {
             });
     }
 
+    public removeAll() {
+        recent
+            .get()
+            .then((actions: Action[]) => {
+                this.remove(actions.map((action: Action) => action.uuid));
+            })
+            .catch((err: Error) => {
+                console.error(`Fail to remove all recent actions: ${err.message}`);
+            });
+    }
+
+    public entryUpdate(): Subscription {
+        return recent.entryUpdate.subscribe(this.reload.bind(this));
+    }
```

Comment:
no needs to share subscription with parent component, because you have `ilc` in state already
```ilc: IlcInterface & ChangesDetector```